server/controllers/PasswordResetController.js

```diff
@@ -0,0 +1,122 @@
+        const passwordResetUrl = `${req.protocol}://${req.headers.host}/api/users/account/password/reset?tokenId=${token}`;
+        const resetEmailMessage = emails.passwordReset(email, passwordResetUrl, user.firstName);
+        const message = 'A password reset link has been sent to your email. Please check your email';
+
+        // Send User Email Method Using SendGrid
+        PasswordResetController.resetProcessEmail(req, res, resetEmailMessage, message, token);
+      })
+      .catch(err => res.status(500).send({ errors: { message: err.message } }));
+  }
+
+  /**
+  * Checks if email exist and sends a reset email if email exit.
+  * @param {object} req the request object
+  * @param {object} res the response object
+  */
+  static verifyPasswordResetToken(req, res) {
+    const decoded = JwtHelper.verifyToken(req.query.tokenId);
+    // A newly generated token to update the password
+    const { email } = decoded;
+    const passwordUpdateToken = JwtHelper.createToken({ email }, 900);
+    return res.status(200).send({
+      status: 'success',
+      message: 'Please use the link below to change your password',
+      token: passwordUpdateToken,
+      redirectUrl: `${req.protocol}://${req.headers.host}/api/users/account/password/reset?tokenId=${passwordUpdateToken}`,
+    });
+  }
+
+  /**
+  * Update user password and sends a confirmation mail on succesful password reset.
+  * @param {object} req the request object
+  * @param {object} res the response object
+  */
+  static updateUserPassword(req, res) {
+    const decoded = JwtHelper.verifyToken(req.query.tokenId);
+    // A newly generated token to update the password
+    const { email } = decoded;
+    const { password } = req.body;
+    const encryptedPassword = bcrypt.hashSync(password, 10);
+
+    User.find({
+      where: {
+        email,
+      }
+    }).then((updatePassword) => {
+      if (updatePassword) {
+        updatePassword.updateAttributes({
+          password: encryptedPassword,
+        }).then((user) => {
+          const resetConfirmMessage = emails.passwordResetConfirmation(email, user.firstName);
+          const message = 'Your Password has been updated successfully! You can login to enjoy stories accross the globe.';
+          // Send User Email Method Using SendGrid
+          PasswordResetController.resetProcessEmail(req, res, resetConfirmMessage, message, 'N/A');
+        });
+      }
+    })
+      .catch(err => res.status(500).send({ errors: { message: err.message } }));
+  }
+
+  /**
+  * Send an email notifying user on successful password reset or email update
+  * @param {object} req the request object
+  * @param {object} res the response object
+  */
+  static resetProcessEmail(req, res, emailMessage, message, token) {
+    Mailer.sendMail(emailMessage)
+      .then((response) => {
+        if (response[0].statusCode === 202) {
+          return res.status(201).json({
+            status: 'success',
+            message,
+            token
```

Comment:
Hi @yomigeek good job with this PR. I noticed that you returned the token you are send ing in the email to the user. This is a security issue as you are effectively giving him the key without him needing to go into his email. I think you should remove it.